table.go

```diff
@@ -124,6 +125,25 @@ func transformFields(m map[string]interface{}) {
 	}
 }
 
+// allFieldValuesAreNullable checks if all the fields passed in contain
+// items that will eventually be marked in Cassandra as null (so we can
+// better predict if we should do an INSERT or an UPSERT). This is to
+// protect against https://issues.apache.org/jira/browse/CASSANDRA-11805
+func allFieldValuesAreNullable(fields map[string]interface{}) bool {
+	for _, value := range fields {
+		rv := reflect.ValueOf(value)
+		switch rv.Kind() {
+		case reflect.Array, reflect.Slice, reflect.Map:
```

Comment:
Just to check — are these the only types which could be nullable? 

I'm trying to see where passing an incorrect type would be filtered out, but I can't see anything that does this? Is it possible for value to be a nil pointer for example?

In `Set()` below ([L177](https://github.com/monzo/gocassa/pull/60/files#diff-cce950b40a3962c50522cd22e55b0b51R177)) a map[string]interface{} with a pointer type (say a nil pointer to a string — `(*string)(nil)`) seems to be passed through and would result in newWriteOp returning an `updateOpType`?

https://play.golang.org/p/lOA15Bdk5v3

Would this be filtered out later by GoCQL as a field of type `*string` would be invalid, or converted to type `string` and therefore `nil -> ""` ?

Reply:
Pointers are filtered out a layer above through `reflect.Indirect` when the [struct is converted into a map](https://github.com/monzo/gocassa/blob/master/reflect/reflect.go) to begin with. 

Also yes, those are the fields that I found to be susceptible to the nullable behaviour.

Reply:
I have [added an explicit test](https://github.com/monzo/gocassa/pull/60/commits/d73135a970d51b27a12c25a28a36383cc3f0cbd7) for this behaviour just to make sure it's codified